waku/nwaku.go

```diff
@@ -330,36 +330,78 @@ const requestTimeout = 30 * time.Second
+	StoreMessageRetentionPolicy string           `json:"storeMessageRetentionPolicy,omitempty"`
+	StoreMessageDbUrl           string           `json:"storeMessageDbUrl,omitempty"`
+	StoreMessageDbVacuum        bool             `json:"storeMessageDbVacuum,omitempty"`
+	StoreMaxNumDbConnections    int              `json:"storeMaxNumDbConnections,omitempty"`
+	StoreResume                 bool             `json:"storeResume,omitempty"`
+	Filter                      bool             `json:"filter,omitempty"`
+	Filternode                  string           `json:"filternode,omitempty"`
+	FilterSubscriptionTimeout   int64            `json:"filterSubscriptionTimeout,omitempty"`
+	FilterMaxPeersToServe       uint32           `json:"filterMaxPeersToServe,omitempty"`
+	FilterMaxCriteria           uint32           `json:"filterMaxCriteria,omitempty"`
+	Lightpush                   bool             `json:"lightpush,omitempty"`
+	LightpushNode               string           `json:"lightpushnode,omitempty"`
+	LogLevel                    string           `json:"logLevel,omitempty"`
+	DnsDiscovery                bool             `json:"dnsDiscovery,omitempty"`
+	DnsDiscoveryUrl             string           `json:"dnsDiscoveryUrl,omitempty"`
+	MaxMessageSize              string           `json:"maxMessageSize,omitempty"`
+	Staticnodes                 []string         `json:"staticnodes,omitempty"`
+	Discv5BootstrapNodes        []string         `json:"discv5BootstrapNodes,omitempty"`
+	Discv5Discovery             bool             `json:"discv5Discovery,omitempty"`
+	Discv5UdpPort               int              `json:"discv5UdpPort,omitempty"`
+	ClusterID                   uint16           `json:"clusterId,omitempty"`
+	Shards                      []uint16         `json:"shards,omitempty"`
+	PeerExchange                bool             `json:"peerExchange,omitempty"`
+	PeerExchangeNode            string           `json:"peerExchangeNode,omitempty"`
+	TcpPort                     int              `json:"tcpPort,omitempty"`
+	RateLimits                  RateLimitsConfig `json:"rateLimits,omitempty"`
+}
+
+type RateLimitsConfig struct {
+	Filter       *RateLimit `json:"-"`
+	Lightpush    *RateLimit `json:"-"`
+	PeerExchange *RateLimit `json:"-"`
+}
+
+func (rlc RateLimitsConfig) MarshalJSON() ([]byte, error) {
+	output := []string{}
+	if rlc.Filter != nil {
+		output = append(output, fmt.Sprintf("filter:%s", rlc.Filter.String()))
+	}
+	if rlc.Lightpush != nil {
+		output = append(output, fmt.Sprintf("lightpush:%s", rlc.Lightpush.String()))
+	}
+	if rlc.PeerExchange != nil {
+		output = append(output, fmt.Sprintf("px:%s", rlc.PeerExchange.String()))
+	}
+	return json.Marshal(output)
+}
+
+type RateLimitUnit string
+
+const Hour RateLimitUnit = "h"
+const Minute RateLimitUnit = "m"
+const Second RateLimitUnit = "s"
+const Millisecond RateLimitUnit = "ms"
+
+type RateLimit struct {
+	Volume int
+	Period int
+	Unit   RateLimitUnit
```

Comment:
Maybe ;P?
```suggestion
	Volume int
	NumTimeUnits int
	TimeUnit   RateLimitUnit
```
Regardless of that, we need to add a comment on every component to make it even clearer

Reply:
Makes sense, thank you!

I changed `Unit` for `TimeUnit` but left `Period` as is, personally `NumTimeUnits` confused my more than `Period`.

I added a comment next to every field explaining what they are